### PR TITLE
Arch arm mpu improve documentation

### PR DESCRIPTION
**Pull request by ioannisg**
Documentation improvements only (an essential clarification about fixed MPU regions' configuration at boot time)

### CODE REVIEW

File: include/arch/arm/cortex_m/mpu/arm_mpu.h [hunk 1]
```diff
@@ -80,7 +80,14 @@ struct arm_mpu_config {
...
 	}
 
-/* Reference to the MPU configuration */
+/* Reference to the MPU configuration.
+ *
+ * It shall be defined and populated for each SOC (in SOC definition). It
+ * shall hold the build-time configuration information for the fixed MPU
+ * regions that are to be enabled upon Kernel initialization. It does not
+ * intend to hold regions that are (dynamically) modified during run-time
+ * operation.
```
Comment (MaureenHelm) [inline include/arch/arm/cortex_m/mpu/arm_mpu.h]: Can we also make it a const?

Comment (ioannisg) [reply include/arch/arm/cortex_m/mpu/arm_mpu.h]: @MaureenHelm , I believe we can do that. Then we would need to change this in all SOCs /Boards that define the boot-time MPU configuration. I opened #10320 for that.

File: include/arch/arm/cortex_m/mpu/arm_mpu.h [hunk 2]
```diff
@@ -80,7 +80,14 @@ struct arm_mpu_config {
 		.attr = _attr, \
 	}
 
-/* Reference to the MPU configuration */
+/* Reference to the MPU configuration.
+ *
+ * It shall be defined and populated for each SOC (in SOC definition). It
```
Comment (dbkinder) [inline include/arch/arm/cortex_m/mpu/arm_mpu.h]: Rather stilted English.  Can this be written as:

    This struct is defined and populated for each SoC (in the SoC definition),
    and holds the build-time configuration information for the fixed MPU
    regions enabled during kernel initialization.  Dynamic regions modified
    during runtime are not kept here.

If that's OK, can you add where these dynamic regions are kept?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: include/arch/arm/cortex_m/mpu/arm_mpu.h (hunk 2 above)
Comment (qianfan-Zhao) [reply include/arch/arm/cortex_m/mpu/arm_mpu.h]: Or can zephyr support dynamic MPU regions un arm soc now?

Comment (ioannisg) [reply include/arch/arm/cortex_m/mpu/arm_mpu.h]: @dbkinder , this is OK. Dynamic regions, however are configured on the fly and are not stored anywhere, at the moment.

**Comment by ioannisg** [comment]
Rebased. @dbkinder , I updated the text based on your suggestion. Pls, re-review, and check my comment about dynamic MPU regions.

**Review by dbkinder — APPROVED** [review]
Ok. Thanks

**Comment by ioannisg** [comment]
recheck

**Review by galak — APPROVED** [review]

**Pull request merged**